evm_arithmetization/src/cpu/kernel/tests/transient_storage.rs

```diff
@@ -0,0 +1,315 @@
+    interpreter.generation_state.registers.is_kernel = true;
+    let slot: U256 = 4.into();
+    interpreter.push(slot);
+    interpreter.push(kexit_info);
+
+    interpreter.run()?;
+
+    let val = interpreter.stack()[0];
+
+    assert_eq!(U256::zero(), val);
+    Ok(())
+}
+
+#[test]
+fn test_many_tstore_many_tload() -> Result<()> {
+    let kexit_info = 0xdeadbeefu32.into();
+
+    let initial_stack = vec![
+        1.into(), // val
+        2.into(), // slot
+        kexit_info,
+    ];
+
+    let mut interpreter: Interpreter<F> = Interpreter::new_with_kernel(0, initial_stack);
+    let gas_limit_address = MemoryAddress {
+        context: 0,
+        segment: Segment::ContextMetadata.unscale(),
+        virt: ContextMetadata::GasLimit.unscale(),
+    };
+    let addr_addr = MemoryAddress {
+        context: 0,
+        segment: Segment::ContextMetadata.unscale(),
+        virt: ContextMetadata::Address.unscale(),
+    };
+
+    interpreter
+        .generation_state
+        .memory
+        .set(gas_limit_address, (10 * 200).into());
+    interpreter.generation_state.memory.set(addr_addr, 3.into());
+
+    let sys_tstore = KERNEL.global_labels["sys_tstore"];
+
+    for i in (0..10) {
+        interpreter.generation_state.registers.program_counter = sys_tstore;
+        interpreter.generation_state.registers.is_kernel = true;
+        let kexit_info = U256::from(0xdeadbeefu32)
+            + (U256::from(interpreter.generation_state.registers.gas_used) << 192);
+        let val: U256 = i.into();
+        let slot: U256 = i.into();
+        interpreter.push(val);
+        interpreter.push(slot);
+        interpreter.push(kexit_info);
+
+        interpreter.run()?;
+        assert_eq!(
+            interpreter.generation_state.registers.gas_used,
+            100 * (i + 1)
+        );
+    }
+
+    let sys_tload = KERNEL.global_labels["sys_tload"];
+
+    for i in (0..10) {
+        interpreter.generation_state.registers.program_counter = sys_tload;
+        interpreter.generation_state.registers.is_kernel = true;
+        let kexit_info = U256::from(0xdeadbeefu32)
+            + (U256::from(interpreter.generation_state.registers.gas_used) << 192);
+        let slot: U256 = i.into();
+        interpreter.push(slot);
+        interpreter.push(kexit_info);
+
+        interpreter.run()?;
+        assert_eq!(
+            interpreter.generation_state.registers.gas_used,
+            100 * (i + 10 + 1)
+        );
+
+        assert_eq!(U256::from(i), interpreter.pop().unwrap());
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_revert() -> Result<()> {
+    // We use a modified kernel with an extra file defining a label
+    // where the `checkpoint` macro from file cpu/kernel/asm/journal/journal.asm
+    // is expanded.
+    let kernel_files: [&str; 152] = array::from_fn(|i| {
+        if i < KERNEL_FILES.len() {
+            KERNEL_FILES[i]
+        } else {
+            include_str!("checkpoint_label.asm")
+        }
+    });
```

Comment:
we can't use this floating 152 around, this is too error-prone. Why can't you just expand it with the new file? It doesn't have to be an array with a const generic, could simply be a vec?

Reply:
> the values are still 0
not `@GAS_WARMACCESS`?

Reply:
I believe it should be 0, because the gas cost is dynamic no? Similar to SSTORE/SLOAD?

Reply:
Ahh, I guess is 0 for all syscalls because the call directly `%charge_gas`